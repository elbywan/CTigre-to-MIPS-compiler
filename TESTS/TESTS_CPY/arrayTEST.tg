type intArray = array of int
and couple = {fst : int, snd : int} in
var x := 0 in




var tab := intArray[2] of (intArray[2] of x:=x-1; x) in

printint(tab);print(" <== address\n");
printint(tab[1]);print(" <== value at [0] (is an address)\n");

printint(tab[0][0]);print(" <== value at [0][0] (is -1)\n");
printint(tab[1][0]);print(" <== value at [0][0] (is -3)")
