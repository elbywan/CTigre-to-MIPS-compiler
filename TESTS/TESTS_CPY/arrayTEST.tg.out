4 <== address
28 <== value at [0] (is an address)
-1 <== value at [0][0] (is -1)
-3 <== value at [0][0] (is -3)
